api/v1alpha1/s3bucket_types.go

```diff
@@ -0,0 +1,132 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// S3BucketParameters define the desired state of a Cloudscale S3 Bucket
+// https://www.cloudscale.ch/en/api/v1#objects-users
+type S3BucketParameters struct {
+	// NameFormat specifies the name of the external S3Bucket instance. The
+	// first instance of the string '%s' will be replaced with the Kubernetes
+	// UID of this S3Bucket. Omit this field to use the UID alone as the name.
+	// +optional
+	NameFormat string `json:"nameFormat,omitempty"`
```

Comment:
Not 100% required at this stage by any means, but it is standard to make optional fields pointer values (i.e. `*string` here). @muvaf wrote up some great documentation on this that directly references the k8s documentation as well https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#pointer-types-and-markers